titan-server/src/organizations/roles.rs

```diff
@@ -121,26 +121,13 @@ pub fn find_org_role_by_user_id(
 /// determines if the user is a standard member or a leader in the
 /// organization, then uses those details as a basis for calculating
 /// the CoC.
-///
-/// ## Deriving the next user in CoC
-/// If a user holds a role in the initial organization, search the
-/// current organization for a user with a lower value for `rank`.
-///
-/// If no such user is found or the user does not hold a leadership
-/// role, set the user's rank to the maximum i32 value, then query
-/// the or parent organization(s) for a user with the highest value
-/// for `rank`. The higher rank value indicates that user holds the
-/// lowest authority in the parent organization.
 pub fn find_user_coc(
     org_id: i32,
     user_id: i32,
     titan_db: &MysqlConnection,
     wcf_db: &MysqlConnection,
     app_config: State<config::AppConfig>
-) -> Result<Vec<models::OrganizationRoleWithUser>, diesel::result::Error> {
-    let mut superiors: Vec<models::OrganizationRoleWithUser> = vec!();
-    let mut internal_org_id = org_id;
-    let mut organization: models::Organization;
+) -> Result<models::ChainOfCommand, diesel::result::Error> {
     let role_res = find_org_role_by_user_id(org_id, user_id, titan_db);
     let mut rank: i32;
```

Comment:
This doesn't actually need to be mutable. Since both instances where this variable is assigned a value is disjoint from one another, the value isn't actually initialized until then. Therefore, it doesn't need to be marked `mut`.